include/nuttx/mm/mm.h

```diff
@@ -263,6 +270,12 @@ struct mm_heap_s
    */
 
   struct mm_freenode_s mm_nodelist[MM_NNODES];
+
+#ifdef __KERNEL__
+  /* Free delay list, for some situation can't do free immdiately */
+
+  struct mm_delaynode_s mm_delaylist;
```

Comment:
change to:
struct mm_delaynode_s *mm_delaylist;
could simplify the code a littel bit

Reply:
I don't think this matters since struct mm_delaynode_s is defined to be only:

    #ifdef __KERNEL__
    struct mm_delaynode_s
    {
      struct mm_delaynode_s *flink;
    };
    #endif

Reply:
A better, more consistent name would probably be mm_delayhead to indicate that it is a head of a list.  sq_queue_t could be used, but that is not really necessary.

Reply:
Ok, let wait guiding provide a minor fix tomorrow.